OpenRA.Mods.Cnc/Traits/Chronoshiftable.cs

```diff
@@ -59,14 +59,18 @@ public Chronoshiftable(ActorInitializer init, ChronoshiftableInfo info)
 			: base(info)
 		{
 			self = init.Self;
-			ReturnTicks = init.GetValue<ChronoshiftReturnInit, int>(info, 0);
-			duration = init.GetValue<ChronoshiftDurationInit, int>(info, 0);
-			Origin = init.GetValue<ChronoshiftOriginInit, CPos>(info, CPos.Zero);
-
-			// Defer to the end of tick as the lazy value may reference an actor that hasn't been created yet
-			var chronosphereInit = init.GetOrDefault<ChronoshiftChronosphereInit>(info);
-			if (chronosphereInit != null)
-				init.World.AddFrameEndTask(w => chronosphere = chronosphereInit.Value(init.World).Value);
+
+			var returnInit = init.GetOrDefault<ChronoshiftReturnInit>(info);
+			if (returnInit != null)
+			{
+				ReturnTicks = returnInit.Ticks;
+				duration = returnInit.Duration;
+				Origin = returnInit.Origin;
+
+				// Defer to the end of tick as the lazy value may reference an actor that hasn't been created yet
+				if (returnInit.Chronosphere != null)
```

Comment:
This check is probably not needed as we will crash later on if no chronosphere is set. So we either remove it or throw an exception here. I tried using `FieldLoader.Require` on those properties but that didn't help either.